templates/telemetry/clients_daily_histogram_aggregates_v1.sql.py

```diff
@@ -0,0 +1,444 @@
+                    UNNEST(histogram_aggregates)
+                ) AS histogram_aggregates
+            )
+        FROM
+            windowed
+        WHERE
+            _n = 1
+    """
+
+    sql_strings[
+        "windowed_clause"
+    ] = f"""
+        SELECT
+            ROW_NUMBER() OVER w1_unframed AS _n,
+            DATE(submission_timestamp) as submission_date,
+            client_id,
+            normalized_os as os,
+            SPLIT(application.version, '.')[OFFSET(0)] AS app_version,
+            application.build_id AS app_build_id,
+            normalized_channel AS channel,
+                {probes_string}
+            FROM deduplicated
+            WINDOW
+                -- Aggregations require a framed window
+                w1 AS (
+                    PARTITION BY
+                        client_id,
+                        DATE(submission_timestamp),
+                        normalized_os,
+                        SPLIT(application.version, '.')[OFFSET(0)],
+                        application.build_id,
+                        normalized_channel
+                    ORDER BY `submission_timestamp` ASC ROWS BETWEEN UNBOUNDED PRECEDING
+                    AND UNBOUNDED FOLLOWING
+                ),
+
+                -- ROW_NUMBER does not work on a framed window
+                w1_unframed AS (
+                    PARTITION BY
+                        client_id,
+                        DATE(submission_timestamp),
+                        normalized_os,
+                        SPLIT(application.version, '.')[OFFSET(0)],
+                        application.build_id,
+                        normalized_channel
+                    ORDER BY `submission_timestamp` ASC
+                )
+    """
+
+    return sql_strings
+
+
+def get_histogram_probes(histogram_type):
+    """Return relevant histogram probes."""
+    project = "moz-fx-data-shar-nonprod-efed"
+    main_summary_histograms = set()
+    process = subprocess.Popen(
+        [
+            "bq",
+            "show",
+            "--schema",
+            "--format=json",
+            f"{project}:telemetry_live.main_v4",
+        ],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+    )
+    stdout, stderr = process.communicate()
+    if process.returncode > 0:
+        raise Exception(
+            f"Call to bq exited non-zero: {process.returncode}", stdout, stderr
+        )
+    main_summary_schema = json.loads(stdout)
+
+    # Fetch the histograms field
+    histograms_field = None
+    for field in main_summary_schema:
+        if field["name"] != "payload":
+            continue
+
+        for payload_field in field["fields"]:
+            if payload_field["name"] == histogram_type:
+                histograms_field = payload_field
+                break
+
+    if histograms_field is None:
+        return
+
+    for histogram in histograms_field.get("fields", []):
+        main_summary_histograms.add(histogram["name"])
```

Comment:
Unless I'm reading this wrong, you're providing a list as the default value to the `.get()`, but in the loop you're treating `histogram` as a dict, which would result in a `TypeError` if the key `fields` didn't exist.

Reply:
good catch!